mmda/parsers/symbol_scraper_parser.py

```diff
@@ -17,51 +17,50 @@
 from collections import defaultdict
 
 from mmda.types.span import Span
-from mmda.types.document import Document, Page, Token, Row, Sent, Block, Text, DocImage
-from mmda.types.boundingbox import BoundingBox
+from mmda.types.box import Box
+from mmda.types.annotation import SpanGroup
+from mmda.types.document import Document
 from mmda.parsers.parser import BaseParser
+from mmda.types.names import *
 
 
 class SymbolScraperParser(BaseParser):
     def __init__(self, sscraper_bin_path: str):
         self.sscraper_bin_path = sscraper_bin_path
 
-    def parse(self, infile: str, outdir: Optional[str] = None, outfname: Optional[str] = None, load_images=True) -> Document:
-        
-        if outdir is None:
-            with tempfile.TemporaryDirectory() as outdir:
-                xmlfile = self._run_sscraper(infile=infile, outdir=outdir)
+    def parse(self, input_pdf_path: str, output_json_path: Optional[str] = None,
+              tempdir: Optional[str] = None, load_images=False) -> Document:
+
+        if tempdir is None:
+            with tempfile.TemporaryDirectory() as tempdir:
+                xmlfile = self._run_sscraper(input_pdf_path=input_pdf_path, outdir=tempdir)
                 doc: Document = self._parse_xml_to_doc(xmlfile=xmlfile)
-            outdir = None
         else:
-            xmlfile = self._run_sscraper(infile=infile, outdir=outdir)
+            xmlfile = self._run_sscraper(input_pdf_path=input_pdf_path, outdir=tempdir)
             doc: Document = self._parse_xml_to_doc(xmlfile=xmlfile)
-       
-        if load_images:
-            doc.load(images=self.load_images(infile))
 
-        if outdir is not None:
-            if outfname is None:
-                outfname = os.path.join(outdir, os.path.basename(infile).replace('.pdf', '.json'))
+        if load_images:
+            raise NotImplementedError(f'Load images?')
 
-            outfile = os.path.join(outdir, outfname)
-            with open(outfile, 'w') as f_out:
+        if output_json_path:
+            with open(output_json_path, 'w') as f_out:
                 json.dump(doc.to_json(), f_out, indent=4)
```

Comment:
Removing `indent=4` can further reduce the storage redundancy -- maybe we can add a todo here?

Reply:
added